framework/src/vectorpostprocessors/VectorPostprocessorData.C

```diff
@@ -29,38 +29,57 @@ VectorPostprocessorData::hasVectorPostprocessor(const std::string & name)
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValue(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values[vpp_name][vector_name];
+  _requested_items.emplace(vpp_name + "::" + vector_name);
 
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values");
-
-  return *pp_val;
+  return getVectorPostprocessorHelper(vpp_name, vector_name, true);
 }
 
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValueOld(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values_old[vpp_name][vector_name];
-
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values_old");
+  _requested_items.emplace(vpp_name + "::" + vector_name);
 
-  return *pp_val;
+  return getVectorPostprocessorHelper(vpp_name, vector_name, false);
 }
 
 
 VectorPostprocessorValue &
 VectorPostprocessorData::declareVector(const std::string & vpp_name, const std::string & vector_name)
 {
-  getVectorPostprocessorValueOld(vpp_name, vector_name);
+  _supplied_items.emplace(vpp_name + "::" + vector_name);
+
+  return getVectorPostprocessorHelper(vpp_name, vector_name, true);
+}
+
+VectorPostprocessorValue &
+VectorPostprocessorData::getVectorPostprocessorHelper(const VectorPostprocessorName & vpp_name, const std::string & vector_name, bool get_current)
+{
+  // Intentional use of RHS brackets on a std::map to do a multilevel retrieve or insert
+  auto & vec_struct = _values[vpp_name][vector_name];
+
+  if (!vec_struct.current)
+  {
+    mooseAssert(!vec_struct.old, "Uninitialized pointers in VectorPostprocessor Data");
+    vec_struct.current = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values");
```

Comment:
Yuck - does this have to be this way for "lazy" binding?  It seems like the restartable data should be declared when the vector is declared.

(Note: I realize that the old code was doing the same)